retry_test.go

```diff
@@ -204,3 +205,158 @@ func TestRoundTripConcurrencySafety(t *testing.T) {
 		assert.GreaterOrEqual(t, res.Delay, initialDelay, res.Res)
 	}
 }
+
+// TestDefaultRetryPolicy_IsErrorRetryable ensures that the IsErrorRetryable
+// method of DefaultRetryPolicy behaves as expected, correctly identifying
+// retryable and non-retryable errors.
+func TestDefaultRetryPolicy_IsErrorRetryable(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    error
+		expected bool
+	}{
+		{
+			name:     "nil error",
+			input:    nil,
+			expected: true,
+		},
+		{
+			name:     "retryable error",
+			input:    errors.New("connection refused"),
+			expected: true,
+		},
+		{
+			name:     "non-retryable error",
+			input:    errors.New("unknown error"),
+			expected: false,
+		},
+	}
+
+	policy := DefaultRetryPolicy{}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
```suggestion
		t.Run(tc.name, func(t *testing.T) {
		    t.Parallel()
```

`t.Parallel()` can also be added for subtests.